typings/index.d.ts

```diff
@@ -1973,7 +1973,10 @@ declare module 'discord.js' {
 
   type Constructable<T> = new (...args: any[]) => T;
   function PartialTextBasedChannel<T>(Base?: Constructable<T>): Constructable<T & PartialTextBasedChannelFields>;
-  function TextBasedChannel<T>(Base?: Constructable<T>): Constructable<T & TextBasedChannelFields>;
+  function TextBasedChannel<T, Ignore extends keyof TextBasedChannelFields = never>(
+    Base?: Constructable<T>,
+    ignore?: Ignore[],
+  ): Constructable<T & Omit<TextBasedChannelFields, Ignore>>;
```

Comment:
```suggestion
  function TextBasedChannel<T, I extends keyof TextBasedChannelFields = never>(
    Base?: Constructable<T>,
    ignore?: I[],
  ): Constructable<T & Omit<TextBasedChannelFields, I>>;
```
wouldn't it be more consistent for the type parameter to be 1 letter (also not sure if I or O for Omit would be better so leaving it as I)

Reply:
Yeah I agree, having it to be one letter would be more consistent. Didn't think about it at the time.